File: library/transform/shadow_bees.py
from library.model.sprite import Sprite
from library.model.sprite_id import SpriteId
from library.model.model import Model


def patch_shadow_bees(context: Model) -> None:
    bees: Sprite = context.sprites[SpriteId.x79_BEE]
    bees.display_allocation = 0
    bees.draw_shadow = True
    bees.statis = True
    bees.boss_damage_sfx = True
    bees.hp = 0x20

File: library/zelda3/features/shadow_bees.rs
use crate::zelda3::model::PaletteIndex;
use crate::zelda3::model::SpriteId;
use crate::zelda3::model::Z3Model;

pub(crate) fn apply_shadow_bees(model: &mut Z3Model) {
    if let Some(bee) = model.sprite_settings.get_mut(&SpriteId::x79_BEE) {
        bee.display_allocation = 0;
        bee.draw_shadow = false;
        bee.statis = true;
        bee.boss_damage_sfx = true;
        bee.hp = 6;
        bee.palette = PaletteIndex::X1;
    }
}
